drrr-infra/src/main/java/com/drrr/infra/fcm/config/FCMConfig.java

```diff
@@ -0,0 +1,43 @@
+package com.drrr.infra.fcm.config;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.messaging.FirebaseMessaging;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+@Configuration
+public class FCMConfig {
+    @Bean
+    FirebaseMessaging firebaseMessaging() throws IOException {
+        ClassPathResource resource = new ClassPathResource("security-storage-infra/firebase/drrr.json");
+
+        InputStream refreshToken = resource.getInputStream();
+
+        FirebaseApp firebaseApp = null;
+        List<FirebaseApp> firebaseAppList = FirebaseApp.getApps();
+
+        if (!firebaseAppList.isEmpty()) {
+            for (FirebaseApp app : firebaseAppList) {
+                if (app.getName().equals(FirebaseApp.DEFAULT_APP_NAME)) {
+                    firebaseApp = app;
+                }
+            }
+        } else {
+            GoogleCredentials googleCredentials = GoogleCredentials.fromStream(refreshToken);
+            FirebaseOptions options = FirebaseOptions.builder()
+                    .setCredentials(googleCredentials)
+                    .build();
+
+            firebaseApp = FirebaseApp.initializeApp(options);
+        }
+
+        return FirebaseMessaging.getInstance(firebaseApp);
+    }
```

Comment:
```suggestion
        if(firebaseAppList.isEmpty()){
            GoogleCredentials googleCredentials = GoogleCredentials.fromStream(refreshToken);
            FirebaseOptions options = FirebaseOptions.builder()
                    .setCredentials(googleCredentials)
                    .build();
            return FirebaseMessaging.getInstance(FirebaseApp.initializeApp(options));
        }
       FirebaseApp firebaseApp = filrebaseApps.stream()
           .filter(app -> Objects.equals(app.getName(), FirebaseAPP.DEFAULT_APP_NAME) 
           .findFirst()
           .orElseThrow(IllegalArgumentException.class);
           
       return FirebaseMessaging.getInstance(firebaseApp);
    }
```
부정조건을 사용 하지 않아도 로직을 구성할 수 있다면 사용하지 않아도 괜찮을 것 같습니다 :)
https://schneide.blog/2014/08/03/dont-ever-not-avoid-negative-logic/

Reply:
좋은 문서 감사합니다! 반영 했습니다